### PR TITLE
Add Vector2

### PR DESCRIPTION
**Pull request by cjpurackal**
#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/geometry/vector.py [hunk 1]
```diff
@@ -79,7 +79,47 @@ def from_coords(
...
+        return cls(rand(shape + (2,), device=device, dtype=dtype))
+
+    @classmethod
+    def from_coords(cls, x: Union[float, Tensor], y: Union[float, Tensor], device=None, dtype=None) -> "Vector2":
+        KORNIA_CHECK(type(x) == type(y))
+        KORNIA_CHECK(isinstance(x, (Tensor, float)))
+        if isinstance(x, float):
+            return wrap(as_tensor((x, y), device=device, dtype=dtype), Vector2)
+        # TODO: this is totally insane ...
+        tensors: Tuple[Tensor, ...] = (x, cast(Tensor, y))
```
Comment (cjpurackal) [inline kornia/geometry/vector.py:117]: Do we really need a cast here? since types of x and y are assured to be the same via checks above.

Comment (johnnv1) [reply kornia/geometry/vector.py:117]: This is a trick to satisfy mypy -- the right way is to use this with overload this function for the type checker know that we expect only floats or only tensors

Comment (cjpurackal) [reply kornia/geometry/vector.py:117]: Got it

**Review by edgarriba — APPROVED** [review]
LGTM. We can deal later with the overloading to make it in a more generic way

**Pull request merged**